server/server.js

```diff
@@ -0,0 +1,41 @@
+/* eslint no-console:0 */
+var loopback = require('loopback');
+var boot = require('loopback-boot');
+var ServiceManager = require('./service-manager');
+
+function server(serviceManager) {
+  var app = module.exports = loopback();
+  app.serviceManager = serviceManager;
+
+  // Bootstrap the application, configure models, datasources and middleware.
+  // Sub-apps like REST API are mounted via boot scripts.
+  boot(app, __dirname);
+
+  app.start = function(callback) {
+    // start the web server
+    app._server = app.listen(function() {
+      var addr = this.address();
+      app.emit('started', addr.port);
+      console.log('Web server listening at port: %s', addr.port);
+      if (callback) return callback(null, addr.port);
+    });
+    return;
+  };
+
+  app.stop = function(callback) {
+    this._server.close(function() {
+      app.emit('stopped');
+      if (callback) callback();
+    });
+  };
+
+  return app;
+}
+
+// start the server if `$ node server.js`
```

Comment:
Schoon started this, intending that all servers be composable since you could require them and add them as handlers, as well as be runnable. Its a good idea in terms of goal, but I don't think the mechanics of this convention are working for us.

We can fix after this PR, but essentially, this mixes require of a module and run of a module in one file... attempting to deduce what was intended. However, the deduction heuristics are just heuristics, and fail regularly. I think it makes more sense for the package, if runnable, to have a top-level `server.js` (this is what `npm start` looks for - in the absence of a start script - and also what `slc run` looks for) and whatever is exported by the package (by package `main:`, or top-level index.js) should never run code, and return a constructor for the app. Perhaps two... a "full ctor" (including the err handler, etc.), and a "minimal" that is *only* the api's exposed by this app' models. That probably means we shouldn't call boot(), but instead break it into two stages: minmal (no middleware), and full-app (with all the middleware).

You've already run into some issues with this, lets talk sometime, and work through what's reasonable.